src/lemonade/tools/server/utils/macos_tray.py

```diff
@@ -0,0 +1,254 @@
+                    menu_item = rumps.MenuItem(
+                        item.text,
+                        callback=(
+                            self._create_callback_wrapper(item)
+                            if item.callback
+                            else None
+                        ),
+                    )
+
+                    # Set enabled state
+                    if not item.enabled:
+                        menu_item.set_callback(None)
+
+                    # Set checked state
+                    if item.checked:
+                        menu_item.state = 1
+                    else:
+                        menu_item.state = 0
+
+                    rumps_items.append(menu_item)
+
+        return rumps_items
+
+    def _create_callback_wrapper(self, item):
+        """Create a callback wrapper that matches our interface."""
+
+        def wrapper(sender):  # pylint: disable=unused-argument
+            if item.callback:
+                item.callback(None, item)
+
+        return wrapper
+
+    def show_balloon_notification(
+        self, title, message, timeout=5000
+    ):  # pylint: disable=unused-argument
+        """
+        Show a notification on macOS using the Notification Center.
+        Falls back to console output if AppleScript fails.
+        """
+        try:
+            # Escape quotes in message and title for AppleScript
+            escaped_title = title.replace('"', '\\"')
+            escaped_message = message.replace('"', '\\"')
+            escaped_app_name = self.app_name.replace('"', '\\"')
+
+            # Use AppleScript to show notification
+            script = (
+                f'display notification "{escaped_message}" '
+                f'with title "{escaped_title}" subtitle "{escaped_app_name}"'
+            )
+            subprocess.run(
+                ["osascript", "-e", script], check=True, capture_output=True, text=True
+            )
+        except FileNotFoundError:
+            # osascript not available, fallback to console
+            print(f"[{self.app_name}] {title}: {message}")
+        except subprocess.CalledProcessError as e:
+            # AppleScript failed, fallback to console
+            print(f"[{self.app_name}] {title}: {message}")
+            print(f"Warning: Failed to show notification via AppleScript: {e}")
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            # Any other error, fallback to console
+            print(f"[{self.app_name}] {title}: {message}")
+            print(f"Warning: Failed to show notification: {e}")
+
+    def exit_app(self, _, __):
+        """Exit the application."""
+        if self.app:
+            rumps.quit_application()
+
+    def run(self):
+        """
+        Run the tray application.
+        """
+        if not RUMPS_AVAILABLE:
+            raise RuntimeError(
+                "rumps library is not available. Install with: pip install 'lemonade-sdk[dev]' "
```

Comment:
```suggestion
                "rumps library is not available. Install with: pip install 'lemonade-idk' "
```

Its not in the dev extra, its in the main install_requires

Side note, there are a lot of "if not rumps available, tell the user to pip install rumps" LoC in this file. Any reason it can't just be checked once during init and then never again?